<launch>
    <arg name="enable_ground_truth" default="true" />
    <!--
    <group if="$(arg enable_ground_truth)">
        <include file="$(find final_aerial_project)/launch/aerial_project_final.launch">
            <arg name="run_controller" value="false"/>
        </include>
    </group>

    <group unless="$(arg enable_ground_truth)">
        <include file="$(find final_aerial_project)/launch/aerial_project_final.launch">
            <arg name="run_controller" value="false"/>
            <arg name="odometry" value="odom_filtered" />
        </include>
    </group>
    -->

    <include file="$(find final_aerial_project)/launch/aruco_detector.launch"/>

    <node pkg="final_aerial_project" type="aruco_pos_node" name="aruco_pos" output="screen">
        <remap from="pose_output" to="/firefly/aruco_pose"/>
        <remap from="fiducial_transforms" to="/ventral/fiducial_transforms"/>

        <param name="base_link_frame"  value="firefly/base_link"/>
        <param name="camera_frame"     value="firefly/camera_ventral_optical_link"/>
        <param name="global_frame"     value="world"/>
        <param name="weighting_scale"  value="0.05"/>
    </node>

    <include file="$(find final_aerial_project)/launch/path_planner.launch"/>

    <include file="$(find final_aerial_project)/launch/estimator.launch"/>

    <include file="$(find final_aerial_project)/launch/mission_planner.launch">
        <arg name="enable_ground_truth" value="$(arg enable_ground_truth)"/>
    </include>

</launch>
